### PR TITLE
Tech debt/isAuthenticatedFlow

### PR DESCRIPTION
**Pull request by mdaley-madetech**
#### Work done
- Added tests around authorisedEstablishmentPermissionCheck
- Refactored authorisedEstablishmentPermissionCheck into smaller functions

#### Tests
Does this PR include tests for the changes introduced?
- [x] Yes
- [ ] No, I found it difficult to test
- [ ] No, they are not required for this change

### CODE REVIEW

File: server/utils/security/isAuthenticated.js [hunk 1]
```diff
@@ -62,242 +64,204 @@ const isAuthorised = (req, res, next) => {
...
-          req.role = claim.role;
-          req.establishment = {
-            id: claim.EstblishmentId,
-            uid: claim.EstablishmentUID,
-            isSubsidiary: false,
-            isParent: false,
-          };
+const reqMatchClaimEstablishment = (establishmentIdIsUID, req, claim) => {
+  if (establishmentIdIsUID && claim.EstablishmentUID === req.params.id) {
+    return (req.establishmentId = claim.EstablishmentUID);
```
Comment (duncanc19) [inline server/utils/security/isAuthenticated.js:85]: Is this supposed to be an equals comparison here to return a boolean?

File: server/utils/security/isAuthenticated.js [hunk 2]
```diff
@@ -62,242 +64,204 @@ const isAuthorised = (req, res, next) => {
...
-          });
-
-          Sentry.setContext('user', {
-            username: req.username,
-            id: req.userUid,
-            ip_address: req.headers['x-forwarded-for'] || req.socket.remoteAddress,
-            role: req.role,
-          });
+  if (claim.EstblishmentId === parseInt(req.params.id)) {
+    return (req.establishmentId = claim.EstblishmentId);
```
Comment (duncanc19) [inline server/utils/security/isAuthenticated.js:89]: And same here?

File: server/utils/security/isAuthenticated.js [hunk 3]
```diff
@@ -62,242 +64,204 @@ const isAuthorised = (req, res, next) => {
...
+  } catch (err) {
+    // failed to find establishment by UUID - being a subsidairy of this known parent
+    console.error(
+      `Failed to find subsidiary establishment (${req.params.id}) for this known parent (${claim.EstblishmentId}/${claim.EstablishmentUID})`,
+    );
+    return res.status(403).send(`Not permitted to access Establishment with id: ${escape(req.params.id)}`);
+  }
+};
+
+const authorisedEstablishmentPermissionCheck = async (req, res, next, roleCheck) => {
```
Comment (duncanc19) [inline server/utils/security/isAuthenticated.js:241]: This is the high level function isn't it? Could we move it towards the top of the file just to make it a bit clearer?

File: server/utils/security/isAuthenticated.js (hunk 1 above)
Comment (mdaley-madetech) [reply server/utils/security/isAuthenticated.js]: No, actually, this should return true/false. Updated.

File: server/utils/security/isAuthenticated.js (hunk 2 above)
Comment (mdaley-madetech) [reply server/utils/security/isAuthenticated.js]: Same as previous.

File: server/utils/security/isAuthenticated.js (hunk 3 above)
Comment (mdaley-madetech) [reply server/utils/security/isAuthenticated.js]: Moved to top for easier reference.

File: server/utils/security/isAuthenticated.js [hunk 4]
```diff
@@ -62,242 +64,204 @@ const isAuthorised = (req, res, next) => {
...
-const authorisedEstablishmentPermissionCheck = async (req, res, next, roleCheck) => {
-  try {
-    const token = getToken(req.headers[AUTH_HEADER]);
-    const Token_Secret = config.get('jwt.secret');
+const audOrISSInvalid = (claim, thisIss) => claim.aud !== config.get('jwt.aud.login') || claim.iss !== thisIss;
 
-    // console.log("WA DEBUG - hasAuthorisedEstablishment token: ", token, Token_Secret)
-    if (token) {
-      const claim = jwt.verify(token, Token_Secret);
+const checkEstablishmentIDIsNotNumber = (claim) => !claim.EstblishmentId || isNaN(parseInt(claim.EstblishmentId));
```
[the remark this answers is not in the document]
Comment (mdaley-madetech) [reply server/utils/security/isAuthenticated.js]: Updated for clarity.

**Review by duncanc19 — APPROVED** [review]

**Pull request merged**